doc/object_wrap.md

```diff
@@ -1,13 +1,384 @@
-## Object Wrap
-
-The ```ObjectWrap``` class can be used to expose C++ code to JavaScript. To do
-this you need to extend the ObjectWrap class that contain all the plumbing to connect
-JavaScript code to a C++ object.
-Classes extending ```ObjectWrap``` can be instantiated from JavaScript using the
-**new** operator, and their methods can be directly invoked from JavaScript.
-The **wrap** word refers to a way to group methods and state of your class because it
-will be your responsibility write custom code to bridge each of your C++ class methods.
-
-You are reading a draft of the next documentation and it's in continuous update so
-if you don't find what you need please refer to:
-[C++ wrapper classes for the ABI-stable C APIs for Node.js](https://nodejs.github.io/node-addon-api/)
+# Object Wrap
+
+The `Napi::ObjectWrap` class is used to expose C++ code to JavaScript. Every C++
+class instance will be "wrapped" by a JavaScript object that is managed by the
+`Napi::ObjectWrap` class. To create a wrapper it's necessary to extend the 
+`Napi::ObjectWrap`class that contains all the plumbing to connect JavaScript code
+with a C++ object. Classes extending `Napi::ObjectWrap` can be instantiated from 
+JavaScript using the **new** operator, and their methods can be directly invoked 
+from JavaScript. The **wrap** word refers to a way of grouping methods and state 
+of the class because it will be necessary write custom code to bridge each of 
+your C++ class methods.
+
+## Example
+
+```cpp
+#include <napi.h>
+
+class Example : public Napi::ObjectWrap<Example> {
+  public:
+    static Napi::Object Init(Napi::Env env, Napi::Object exports);
+    Example(const Napi::CallbackInfo &info);
+
+  private:
+    static Napi::FunctionReference constructor;
+    double _value;
+    Napi::Value GetValue(const Napi::CallbackInfo &info);
+    Napi::Value SetValue(const Napi::CallbackInfo &info);
+};
+
+Napi::Object Example::Init(Napi::Env env, Napi::Object exports) {
+    Napi::HandleScope scope(env);
```

Comment:
Is the handle scope needed here? I'm not sure but worth checking

Reply:
No it's not necessary I will remove it. (I tested the example without handle scope).